### ISSUE TITLE
After extract_tables some values have <NEST TABLE> 

### ISSUE THREAD

**Issue by Rustemhak** [dothinking/pdf2docx#179]
Hello. After application of function `extract_tables`  in some lists I get values `<NEST TABLE>`. Is it possible to extract data from `<NEST TABLE>`? If necessary, I can give an example pdf file, where it is.

**Comment by macanv** [comment]
感谢这么好用的工具库
我也遇到同样的问题，表格的单元格内容有时会被解析成<NEST TABLE>， 我发现在表格背景色为灰色的时候很大概率会出现这种问题

pdf中的表格：
<img width="784" alt="image" src="https://user-images.githubusercontent.com/16610116/233231404-bf14f287-4fe6-4b0d-b3db-2ea9dd7d4686.png">

debug：
<img width="1103" alt="image" src="https://user-images.githubusercontent.com/16610116/233231446-46888bf6-1418-4ff1-adb8-22ec691aa9de.png">

请问有什么建议，可以改进么？

**Comment by fefefefefefe** [comment]
同问

**Comment by Rustemhak** [comment]
@macanv Yes, I also noticed that one of the reasons why `<NEST TABLE>` appears when other colors are present in the table. I sometimes had a yellow color in my table because of a comment left in a pdf document, so if possible, I removed all such comments manually. Nevertheless, there are still documents where there is a `<NEST TABLE>` for another reason, I just logged that I did not extract everything from the table.

**Comment by Rustemhak** [comment]
Yes, I also noticed that one of the reasons why `<NEST TABLE>` appears when other colors are present in the table. I sometimes had a yellow color in my table because of a comment left in a pdf document, so if possible, I removed all such comments manually. Nevertheless, there are still documents where there is a `<NEST TABLE>` for another reason, I just logged that I did not extract everything from the table.

**Comment by icarusxxy** [comment]
Is there any update on this issue?
I also encounter this when parsing a table with background color.
See the testing pdf file and the result screenshot below:
![image](https://github.com/user-attachments/assets/627a275a-c597-41c9-9a42-ea0a7d5e5e61)

every even row's(with blue background) header will be parse as `<NEST TABLE>`
[test.pdf](https://github.com/user-attachments/files/16230918/test.pdf)

**Comment by vivekagr** [comment]
@icarusxxy This modification in table/Cell.py worked for me, although it's not perfect, but I'd rather have some text than `<NEST TABLE>`.

```
    @property
    def text(self):
        '''Text contained in this cell.'''
        if not self: return None
        # NOTE: sub-table may exists in

        text = []

        for block in self.blocks:
            if block.is_text_block:
                text.append(block.text)
            elif block.__class__.__name__ == "TableBlock":
                text.append(''.join(flatten(block.text)))
            else:
                logging.warning(f"Found {block.__class__.__name__} - {block.text}")
                text.append("<NEST TABLE>")

        return '\n'.join(text)
```

Although, I eventually moved to using pdfplumber as it's working better for documents I am working with.

**Comment by sterling-swygert** [comment]
@vivekagr is it possible to suggest the change to the authors?

**Comment by sterling-swygert** [comment]
I guess one can just monkey patch it. That's what I ended up doing.
